ods-devenv/scripts/run-on-aws.sh

```diff
@@ -91,45 +100,54 @@ if ! which jq &> /dev/null; then
 fi
 
 if [[  -z "$keypair" ]]; then
-    echo "--key-name not given, setting to default edp_dublin_keypair."
-    keypair=edp_dublin_keypair
+    echo "--keypair not given, setting to default: ${default_keypair}"
+    keypair="${default_keypair}"
+fi
+
+if [[  -n "$profile" ]]; then
+    profile_flag="--profile=${profile}"
 fi
 
 target_git_ref="${target_git_ref:-master}"
 
+# If no host is given, we need to launch one now, unless we have been passed
+# an instance ID, in which case we only need to find its IP.
 if [[ -z "${host}" ]]; then
   if [[ -z "${instance_id}" ]]; then
+    echo "Neither --host nor --instance-id specified, launching new instance."
     if [[ -z "${security_group_id}" ]] && [[ -z "${iam_instance_profile}" ]]; then
-      echo "Neither --security-group-id nor --iam-instance-profile specified. Using default sg-006935bec03a154a1"
-      security_group_id=sg-006935bec03a154a1
-    fi
-    if [[ -z "${keypair}" ]]; then
-      echo "ERROR: --key-name not given."
-      exit 1
+      echo "Neither --security-group-id nor --iam-instance-profile specified. Using default SG: ${default_securitygroup}"
+      security_group_id="${default_securitygroup}"
     fi
     if [[ -z "${ami_id}" ]]; then
-        if [[ -n "${install}" ]]
-        then
-            ami_id=$(aws ec2 describe-images \
-                --owners 275438041116 \
-                --filters "Name=name,Values=import-ami-*" "Name=root-device-type,Values=ebs" "Name=tag:Name,Values=CentOS*" \
+        if [[ -n "${install}" ]]; then
+            ami_id=$(aws ${profile_flag} ec2 describe-images \
+                --owners ${ami_owner} \
+                --filters "Name=name,Values=CentOS*" \
                 --query 'Images[*].{ImageId:ImageId,CreationDate:CreationDate}' | jq -r '. |= sort_by(.CreationDate) | reverse[0] | .ImageId')
+            if [[ -z "${ami_id}" ]] || [[ "null" == "${ami_id}" ]]; then
+                echo "Did not find vanilla CentOS AMI"
+                exit 1
+            fi
             ec2_instance_name="ODS in a box Install (${target_git_ref}) $(date '+%Y-%m-%d %H:%M:%S')"
-            echo "You are in install mode using CentOS 7 image ${ami_id}."
+            echo "You are in install mode using a vanilla CentOS 7 image ${ami_id}."
```

Comment:
it's actually a custom CentOS 7 2003 AMI for the ODS Box

Reply:
#988 :(